101-azure-machine-learning-service-workspace/azuredeploy.json

```diff
@@ -0,0 +1,123 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the Azure Machine Learning service workspace."
+            }
+        },
+        "location": {
```

Comment:
looks like a number of resources in this have limited location availability - need to set allowed values for the union of those.

Reply:
When the workspace is created, a managed system identity is created and associated with it. When the workspace is deleted, the managed system identity is also removed. It's required by the workspace.

Reply:
Apparently the advice I got on that was incorrect. Checking now, it looks like some of the resources are allowed outside of the region. Specifically looks like app insights. Adding another regional list for that.

Is this something that we can set automatically? Or will we have to prompt the user for the location of each resource?

Reply:
What about an IF statement? There's only two locations that aren't available, east us 2 and west central us. Using an IF, I can split the difference and select south central us for both of those for the app insight resource.